test/querystring.test.js

```diff
@@ -134,4 +134,42 @@ describe('JsonClient', function () {
             return done();
         });
     });
+
+
+    it('should prefer default query string in url over query options',
```

Comment:
Isn't this test ensuring the opposite? It seems we pass `foo=bar` via the url, but we test that `req.query` equals to `foo=baz`.

Reply:
Also, there seems to be already a test that does what that tests describes at https://github.com/restify/clients/pull/172/files#diff-fdb58f09f3d3e28b106e34706b336baaR116? However both implementations seem to have a different behavior. The one above tests what it says it does (it tests that the `query` option is effectively ignored), but this one does something different.